slotsmanager.c

```diff
@@ -82,6 +82,9 @@ slots_fork(int parent, int ptype)
             endrun(1, "Assertion Failure more PI than available slots : %d > %d\n",P[child].PI, SlotsManager->info[ptype].maxsize);
         }
 
+        /*Remember to zero the new slot*/
+        memset(BASESLOT(child), 0, SlotsManager->info[ptype].elsize);
```

Comment:
This is right. But what about calling the loop body of this (refactoring it out as a function first):
https://github.com/rainwoodman/MP-Gadget/blob/master/slotsmanager.c#L479

Reply:
I ran into this when writing the test case. So basically whenever a slot is attached to a particle it shall be initialized.

Reply:
I left the loop body intact because we don't want to memset the slot to zero in slots_setup_id: we just read data into those particles.

Reply:
I don't think the data is read in yet when the loop is called.